whmcs_int_test.go

```diff
@@ -0,0 +1,314 @@
+package whmcsgo
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/jinzhu/now"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/stretchr/testify/assert"
+)
+
+type tHelper interface {
+	Helper()
+}
+
+/*
+Integration tests to be run with an active development instance of WHMCS.
+
+The following environment variables are required to run the tests:
+export WHM_IDENT="put your api identity key here"
+export WHM_SECRET="put your api secret key here"
+export WHM_ACCESS="put your api access key here"
+export WHM_URL="http://localhost:####/"
+export WHM_PAYMENTMETHOD="payment method setup in dev env"
+
+Prerequisites to running the tests:
+- Working instance of WHMCS
+- Settings -> API Credentials - Created API Role with appropriate access and create credentials
+- Settings -> Payment Gateways - Atleast one payment gateway must be selected
+*/
+
+type Config struct {
+	URL           string `default:"http://localhost:3000/"`
+	Access        string `default:"RAND0MT3STK3Y"`
+	Ident         string `default:"JnbGfwNUq1CIHxhEoqRbMKb084gcvwwz"`
+	Secret        string `default:"Wx9Lqeqe0Os0paUUDtbc37k89qfpqdvZ"`
+	PaymentMethod string `default:"testPaymentMethod"`
+}
+
+var whmcsConfig *Config
+var testUser = "testdude@divisia.io"
+
+// Import environment variables
+func init() {
+	whmcsConfig = &Config{}
+	if err := envconfig.Process("WHM", whmcsConfig); err != nil {
+		panic(err)
+	}
+}
+
+// Creates a test product
+func createTestProduct(whmcs *Client) (*int, error) {
+	var (
+		prod Product
+	)
+	_, response, err := whmcs.Products.AddProduct(
+		map[string]string{
+			"name": "TestProduct", "gid": "1",
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("AddProduct Failed: %w", err)
+	}
+	json.Unmarshal([]byte(response.Body), &prod)
+
+	if response.StatusCode == 201 || response.StatusCode == 200 {
+		fmt.Printf("Created test product. ProductID: %d\n", prod.Pid)
+		return &prod.Pid, err
+	} else {
+		return nil, fmt.Errorf("error, AddProduct returned status of: %d\n", response.StatusCode)
```

Comment:
error, is superfluous
```suggestion
		return nil, fmt.Errorf("AddProduct returned status of: %d", response.StatusCode)
```